app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -20,6 +20,10 @@ class PuppetclassLookupKey < LookupKey
 
   scope :smart_class_parameters, -> { joins(:environment_classes).readonly(false) }
 
+  def can_edit?
+    User.current.allowed_to?(:edit_external_parameters)
```

Comment:
This checks if the user has that permission, but what if it is only limited to certain parameters?